pkg/api/index.go

```diff
@@ -691,12 +691,16 @@ func (hs *HTTPServer) setIndexViewData(c *models.ReqContext) (*dtos.IndexViewDat
 		return nil, err
 	}
 
-	// Read locale from accept-language
-	acceptLang := c.Req.Header.Get("Accept-Language")
+	// Set locale to the preference, otherwise fall back to the accept language header.
+	// In practice, because the preference has configuration-backed default, the header
+	// shouldn't frequently be used
+	acceptLangHeader := c.Req.Header.Get("Accept-Language")
 	locale := "en-US"
 
-	if len(acceptLang) > 0 {
-		parts := strings.Split(acceptLang, ",")
+	if hs.Features.IsEnabled(featuremgmt.FlagInternationalization) && prefs.JSONData.Locale != "" {
+		locale = prefs.JSONData.Locale
+	} else if len(acceptLangHeader) > 0 {
+		parts := strings.Split(acceptLangHeader, ",")
 		locale = parts[0]
```

Comment:
Your understanding is correct, and it's something I've been trying to think through the impacts of.

I'm trying to figure out exactly what this locale field is currently used for (seems to be some date formatting things?), and how big of a deal it is to change this.

I've created this issue to sort it out https://github.com/grafana/grafana/issues/51133

Reply:
I believe it might be used by the old graph panel (I haven’t refreshed my understanding here) to determine whether month views should be MM/DD or DD/MM and some smaller things like that.

Tracking is great, let’s get back to this in the future 👍